docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +278,229 @@ Note that:
+  * listen over MQTT for configuration updates and requests,
+  * download and upload on demand the configuration files,
+  * notify the progress of the configuration updates to the main device via MQTT.
+
+For each kind of child device such a specific client has to be implemented
+and installed on the child-device hardware (To be precise, most often on the child device,
+but not necessarily as, for a child device which software cannot be altered,
+one might have to install a protocol adaptor on the main device).
+
+Here is the protocol that have to be implemented by child-device configuration client.
+This protocol covers 3 interactions:
+* Receiving configuration file updates from thin-edge
+* Sending to thin-edge current configuration files
+* Notifying the list of configuration files to thin-edge
+
+### The child device connect the main thin-edge device
+
+From a TCP point of view, the child devices act as clients
+and all the connections to thin-edge are established by the child devices.
+* Thin-edge opens two ports for MQTT and HTTP over the local network.
+  These ports are controlled on the main device `tedge config`:
+  * `mqtt.external.bind_address`
+  * `mqtt.external.port`
+  * `http.external.port`
+* The child devices must know the main device IP address.
+  * This is the address set for `mqtt.external.bind_address` on the main device.
+  * For the very specific case, where the software for the child device run on the main device,
+    this connection address can be the `localhost`.
+* On start, the child-device specific software for configuration management,
+  must connect over MQTT to thin-edge device.
+  * It must use a session name that is unique on the thin-edge bus.
+    `"$CHILD_DEVICE_ID/configuration"` is the recommendation.
+  * It must subscribe to `tedge/$CHILD_DEVICE_ID/commands/req/#` to receive requests.
+  * It has to publish responses under `tedge/$CHILD_DEVICE_ID/commands/res/#`.
+* On demand, the child-device software has to send HTTP requests to thin-edge.
+  * These requests are REST requests (GET/PUT/DELETE) to exchange files.
+  * The urls are forged by thin-edge telling the child devices
+    where to get and put configuration files.
+* Currently, all these connections are done without TLS support.
+  * Hence, the main device must be configured so these MQTT and HTTP ports
+    are only open on a local trusted network.
+
+### The child device downloads configuration updates on notification
+
+When a configuration file for a child device
+is received from the cloud by the `c8y_configuration_plugin` daemon,
+the configuration plugin manages the transfer of this file to the child device
+and notifies the cloud on the progress of this configuration update operation.
+
+1. On reception of a configuration update for a child device named `$CHILD_DEVICE_ID`
+   of a file `{ path = $PATH, type = $TYPE }` defined in `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` 
```

Comment:
To avoid breaking the flow of the document, would be better to put an example file `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` onto the top of that section, that can be used/referenced in all cases (config upload, config download, config list).